### PR TITLE
[Bug] Add the zstd to paimon-format artifact

### LINKED ISSUE #1496: [Bug] zstd compress is not available
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

master

### Compute Engine

format

### Minimal reproduce step

use compress zstd
 
image

### What doesn't meet your expectations?

do not work

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Aitozi**
### Purpose

Linked issue: close #1496

### CODE REVIEW

File: paimon-format/pom.xml [hunk 1]
```diff
@@ -298,6 +298,10 @@ under the License.
                                     <include>org.apache.parquet:parquet-format-structures</include>
                                     <include>org.apache.parquet:parquet-jackson</include>
                                     <include>commons-pool:commons-pool</include>
+
+                                    <!-- compress -->
+                                    <include>com.github.luben:zstd-jni</include>
+                                    <include>org.xerial.snappy:snappy-java</include>
```
Comment (Aitozi) [inline paimon-format/pom.xml:304]: The JNI class should not be relocated: https://github.com/luben/zstd-jni#limitations

File: paimon-format/pom.xml [hunk 2]
```diff
@@ -298,6 +298,10 @@ under the License.
                                     <include>org.apache.parquet:parquet-format-structures</include>
                                     <include>org.apache.parquet:parquet-jackson</include>
                                     <include>commons-pool:commons-pool</include>
+
+                                    <!-- compress -->
+                                    <include>com.github.luben:zstd-jni</include>
```
Comment (JingsongLi) [inline paimon-format/pom.xml:303]: please update NOTICE file too.

File: paimon-format/pom.xml (hunk 1 above)
Comment (JingsongLi) [inline paimon-format/pom.xml:304]: Do we need snappy? I think hadoop jars already have snappy?

Comment (Aitozi) [reply paimon-format/pom.xml]: Yes, I just think we should not rely on the hadoop jars. Removed now, we can add back if necessary.

File: paimon-format/pom.xml (hunk 2 above)
Comment (Aitozi) [reply paimon-format/pom.xml:303]: Updated

File: paimon-format/src/main/resources/META-INF/NOTICE [hunk 3]
```diff
@@ -26,6 +26,7 @@ This project bundles the following dependencies under the Apache Software Licens
 - org.apache.parquet:parquet-format-structures:1.12.3
 - org.apache.parquet:parquet-jackson:1.12.3
 - commons-pool:commons-pool:1.6
+- com.github.luben:zstd-jni:1.15.0-1
```
Comment (JingsongLi) [inline paimon-format/src/main/resources/META-INF/NOTICE:29]: This is not apache license: https://github.com/luben/zstd-jni/blob/master/LICENSE

Comment (Aitozi) [reply paimon-format/src/main/resources/META-INF/NOTICE]: fixed

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**